cmd/kubeadm/app/apis/kubeadm/types.go

```diff
@@ -422,3 +420,11 @@ type HostPathMount struct {
 	// PathType is the type of the HostPath.
 	PathType v1.HostPathType
 }
+
+// ObjectMeta is a cut down version on metav1.ObjectMeta with the bare minimum that is sensible in
+// kubeadm terms and satisfies the requirements of Kustomize.
+// The original metav1.ObjectMeta proved too large and some unneeded fields broke a bunch of tests (like the fuzzer).
```

Comment:
I think that we should document why we are creating our object meta as in the PR description
> It contains way too many fields, that are not relevant to kubeadm, it clobbers
the structs in which it's embedded with unnecessary fields and it generates a
lot of unnecessary output when marshalling types to YAML (such as always nil
`DeletionTimestamp`) 

Otherwise, it will be not clear why not we considered to fix failing tests...